### ISSUE TITLE
A few train-related questions

### ISSUE THREAD

**Issue by larjung** [peiyunh/tiny#28]
Hi Peiyun,
Thank you so much for sharing with us your awesome work!
While I've been experimenting with training this net and analyzing your train code, a few points have come up on which i would be very happy to hear your thoughts:

1.  You chose summing the heatmap predictors of conv3&4 other than, say, concatenating the features then learning a single predictor over the concatenated feature. Have you tried other alternatives over sum? Many works on CNN object detection use concatenation for multi-scale features so your opinion is interesting.

2. If I'd want to use only single-scale prediction at test time (e.g. no budget for pyramid, especially not x2 res), and to re-train for this scenario, which configuration of anchor boxes would you suggest me to use for train? Should the unscaled version of your given clusters be good for this, or maybe doing clustering from scratch? I think in this scenario the A/B pattern is no longer desired.

3. The fixed OHEM threshold of 0.03.  This value seems to have a very big impact on the net's convergence. Setting it to zero (i.e. no OHEM) causes the training loss to explode. Setting it to a  smaller number hurts the detection quality.  It seems very delicate. Empirically it filters on average 99.9% of the negatives and 10-70% of the positives per batch. Any insights on how this number was chosen? Also, in the paper the effect of it on mAP is almost negligible, while in reality I'm seeing quite the opposite.

4. What is the rationale for choosing x10 smaller LR for conv3 predictor over conv4?

5. Why are the weights for the predictor conv layers zero-initialized and not randomly initialized as usual? It's not very common to see zero init for conv weights.

Thank you for your patience reading this,
And again, great work!

**Comment by peiyunh** [comment]
1. Given a linear predictor, concatenating features is equivalent to predicting based on individual feature and summing them up. 
2. I would run the detector with an image pyramid but without 2X upsampling. If you give up 2X, it should only hurt the performance on tiny-sized faces. 
3. That is strange. What I saw when I add hard negative mining into our detector is that results look qualitatively much nicer because of less false positives, but it does not yield much improvement in the final accuracy. 
4. This is what I learned from how FCN is trained. I believe they use x100 smaller learning rate when training FCN8s at once but I found x10 works fine. 
5. I believe there are papers initialize predictor with zero weights, such as FCN.

**Comment by thealchemist-x** [comment]
Hi @peiyunh 

Thanks for your amazing work. I'm new to deep learning and am interested to pick up small faces. If I may ask a simple question on the training dataset:

Q) If I have generally large faces in my datasets (example, a face occupying 120 pixels x 70 pixels from a larger image resolution of 640 x 480), do I have to crop the face out and resize to say 30 pixels x 30 pixels for the region of interest and crop the face together with the context and resize to say 60 pixels x 60 pixels before training?

Thank you for your guidance!

**Comment by peiyunh** [comment]
Your idea makes sense and should fit better into a two-stage architecture, such as Faster RCNN.
